packages/react-router/src/useBlocker.tsx

```diff
@@ -3,25 +3,83 @@ import { useRouter } from './useRouter'
 import type { BlockerFn } from '@tanstack/history'
 import type { ReactNode } from './route'
 
+type BlockerResolver = {
+  status: 'idle' | 'blocked'
+  proceed: () => void
+  reset: () => void
+}
+
+type BlockerOpts = {
+  blockerFn?: BlockerFn
+  condition?: boolean | any
+}
+
 export function useBlocker(
```

Comment:
Maybe this needs to be a function overload?

```tsx
export function useBlocker(blockerFn: BlockerFn, condition?: boolean | any): BlockerResolver;
export function useBlocker(blockerOpts: BlockerOpts): BlockerResolver {
  // ...
}
```

And probably should depreciate the old syntax.

Reply:
Done! I also updated the docs with the new API